### PR TITLE
Show the patch version number 

### PR DESCRIPTION
**Pull request by paulmelis**
in places where a version is shown or returned (including RPC).

### CODE REVIEW

**Comment by cryptocode** [comment]
This may be a breaking change for clients depending on the old format. Sounds like there should be a request parameter to indicate patch.

**Comment by paulmelis** [comment]
You mean the RPC return value? Yeah, I was wondering about that

**Comment by cryptocode** [comment]
It's possible to imagine exchange scripts checking the cmdline --version too, but in that case the option is probably too new to be an issue. I'm more worried about the output of RPC commands changing.

**Comment by argakiig** [comment]
The CLI version was added recently, but the RPC one, what about adding an optional flag to show patch version, but default would be to maintain current behavior?

**Comment by paulmelis** [comment]
I added a "patch" parameter, so the returned version number stays compatible with the current string. 

Interestingly, the RPC docs are currently not correct, as they show the patch version being returned when it isn't:

```
{  
  "rpc_version" : "1",
  "store_version" : "2",
  "node_vendor" : "RaiBlocks 7.5.0"
}
```

**Comment by paulmelis** [comment]
Crap, the test is now wrong

**Comment by paulmelis** [comment]
Failed tests look unrelated to this commit?

**Comment by argakiig** [comment]
Failed due to formatting,

```
Code formatting differs from expected - please run ci/clang-format-all.sh```

File: rai/node/rpc.cpp [hunk 1]
```diff
@@ -3126,10 +3126,24 @@ void rai::rpc_handler::unchecked_keys ()
 
 void rai::rpc_handler::version ()
 {
+	bool patch (false);
+	boost::optional<bool> patch_optional (request.get_optional<bool> ("patch"));
+	if (patch_optional.is_initialized ())
+	{
+		patch = patch_optional.get ();
+	}
+
```
Comment (cryptocode) [inline rai/node/rpc.cpp]: All of that can be reduced to `bool patch = request.get<bool> ("patch", false);`

Comment (argakiig) [reply rai/node/rpc.cpp]: I agree I think this would be a good idea at least here @clemahieu thoughts?

**Comment by clemahieu** [comment]
A while ago I removed the patch number from a lot of these places. My reasoning was to try and adopt the versioning scheme by Firefox where they stick with increasing major numbers instead of a long string of dot-separated minor and patch numbers. 

What do you think of using the major number and then maybe the first 6 digits of the sha of the commit it’s built on?

**Comment by paulmelis** [comment]
My reason for this pull request was that the 10.0.1 release didn't identify itself as such, possibly leading to confusion w.r.t incorrect balances. I'd say being able to tell exactly which version you're running should be the main goal here, independent of how the version number looks.

**Comment by androm3da** [comment]
My mistake, then -- I should've made it `10.1` instead.

**Comment by androm3da** [comment]
Instead of accepting this PR, I have noted "don't use the patch number, only major/minor" in a set of items to be gathered into a release process checklist (#628 ).

Please reopen if you think it's still valid.

**Pull request closed without merging**